### PR TITLE
map screen for vehicle interception

### PR DESCRIPTION
**Pull request by calvinschmidt**
What is done here:

- set marker at destination
- track user location
- follow the user location
- get direction from current location to destination and draw it on the map
- show confirm arrival button if current location is closer than 20 meters to destination

### CODE REVIEW

File: screens/HomeScreen.js [hunk 1]
```diff
@@ -82,16 +89,16 @@ const styles = StyleSheet.create({
     fontSize: 32,
     fontWeight: '600',
     color: '#fefefe',
-    textAlign: "center",
+    textAlign: 'center',
   },
-  mapButton: {
+  mapContainer: {
```
Comment (nschoen) [inline screens/HomeScreen.js:93]: This does not belong to this PR, but you can fix it quickly. On Android the radius for the MapMarker does not seem to work. The mapContainer Component should have the `radius` and a `overflow: 'hidden'` style set.

File: components/MapRoute.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+      destinationReached: false,
+      isMapReady: false,
+    };
+
+    this.apikey = appConfig.expo.android.config.googleMaps.apiKey;
+    // bind this in constructor so state can be set in these methods
+    this.getLocation = this.getLocation.bind(this);
+    this.getDirections = this.getDirections.bind(this);
+    this.checkUserLocation = this.checkUserLocation.bind(this);
+    this.animateToCoordinates = this.animateToCoordinates.bind(this);
```
Comment (nschoen) [inline components/MapRoute.js]: I guess Santhos and you already talked about that. You can use arrow-functions to implicitly bind the function to this.

File: components/MapRoute.js [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    if (focusedLocation && latitude && longitude) {
+      this.map.animateToRegion({
+        ...focusedLocation,
+        latitude: latitude,
+        longitude: longitude,
+      });
+    }
+  }
+
+  renderConfirmalButton() {
```
Comment (nschoen) [inline components/MapRoute.js:212]: You could create another Component e.g. ConfirmArrivalButton to make this component a bit more compact. On the other hand, I will create a custom Button Component in my next PR. We could use it to refactor this Button later.

File: helpers/api.js [hunk 4]
```diff
@@ -5,16 +5,17 @@ import fakeApi from './fakeApi';
 const BASE_URL = Api.host;
 
 const request = (method, url, bodyParams, queryParams) => {
-  if (global.devSettings.settings.get('fakeApi')) {
-    const response = fakeApi(method, url, bodyParams, queryParams);
+  if (global.devSettings.settings.get('fakeApi') && url === '/authenticate') {
+    const response = fakeApi(method, '/session', bodyParams, queryParams);
```
Comment (nschoen) [inline helpers/api.js]: Why not just changing the path in fakeApi to `/authenticate`? In that way we can still fake all api calls. And maybe we should set the default value for `fakeApi` in the `DevSettings` store to `false` as we are working with the actual backend anyways.

**Review by nschoen — COMMENTED** [review]
Looks very good, thanks! I just ran into a few issues with Android.

**LocationService**
The following request to the Location API throws an error (rejects the promise) when the Location Service is not activated:
```
const {{ coords }} = await Location.getCurrentPositionAsync({
      enableHighAccuracy: true,
});
```

To check if the location service is enabled you could use:
```
const { locationServicesEnabled } = await Location.getProviderStatusAsync();
```

An android user can be asked to enable them by creating an intent:
```
import { IntentLauncherAndroid } from 'expo';
// ...
IntentLauncherAndroid.startActivityAsync(
  IntentLauncherAndroid.ACTION_LOCATION_SOURCE_SETTINGS
);
```
or just pop up a notifications that the user should enabled GPS/LocationService for now.

**API Key Configuration**
Another problem occurred with the API. For testing I created my own API key, where I had to enable "Maps SDK for Android" to get it to work on Android.

File: screens/HomeScreen.js (hunk 1 above)
Comment (calvinschmidt) [reply screens/HomeScreen.js:93]: Thanks! Done.

File: components/MapRoute.js (hunk 2 above)
Comment (calvinschmidt) [reply components/MapRoute.js]: Thanks! Done.

File: components/MapRoute.js (hunk 3 above)
Comment (calvinschmidt) [reply components/MapRoute.js:212]: I let you refactor it. We should definitely have a custom button component.

File: helpers/api.js (hunk 4 above)
Comment (calvinschmidt) [reply helpers/api.js]: Done. I set `fakeApi ` to `false`.

**Review by nschoen — APPROVED** [review]
